src/main/java/de/retest/recheck/RecheckAdapter.java

```diff
@@ -24,21 +26,34 @@ default RecheckAdapter initialize( final RecheckOptions opts ) {
 	/**
 	 * Returns {@code true} if the given object can be converted by the adapter.
 	 *
-	 * @param toVerify
-	 *            the object to verify
+	 * @param toCheck
+	 *            the object to check
 	 * @return true if the given object can be converted by the adapter
 	 */
-	boolean canCheck( Object toVerify );
+	boolean canCheck( Object toCheck );
 
 	/**
 	 * Convert the given object into a {@code RootElement} (respectively into a set of {@code RootElement}s if this is
 	 * sensible for this type of object).
 	 *
-	 * @param toVerify
-	 *            the object to verify
+	 * @param toCheck
+	 *            the object to check
 	 * @return The RootElement(s) for the given object
 	 */
-	Set<RootElement> convert( Object toVerify );
+	Set<RootElement> convert( Object toCheck );
+
+	/**
+	 * Return some metadata with respect to the checked object. For e.g. a selenium driver, this could be things like
+	 * browser name and version.
+	 *
+	 * @param toCheck
+	 *            the object to check
+	 *
+	 * @return The meta data for the given object
+	 */
+	default Map<String, String> retrieveMetadata( final Object toCheck ) {
+		return new HashMap<>();
```

Comment:
I would use `Collections.emptyMap()` here, since we do not need mutability.

Reply:
An empty map can't be added elements to. Since we do not ensure or enforce immutability of this result, we can't use empty map here.

Reply:
It is not possible to call default methods from overwritten methods, so only the result counts. We do not add elements to this, thus we do not need mutability after `retrieveMetadata` is called. I do not understand your logic here.